### PR TITLE
Add scoring logic for users and teams

### LINKED ISSUE #166: Scoring Logic
We need to design and implement the scoring logic for the teams. This needs to be added to the appropriate views.

General idea: Teams will get points for the percentage decrease in energy usage they have over the previous month. We had some conversations about potential variations and ideally we'd have usage information from the previous year but that is not easy to do until we get Green Button access.

Teams should be ranked on a monthly basis and receive a monthly e-mail update or other notification.

### PR DESCRIPTION
**Pull request by mzagaja**
- Calculates a score for a user based on last two bills
- Calculates a score for team based on all members with at least two bills

Resolves #166

### CODE REVIEW

**Comment by matthewvedder** [comment]
The code is so elegant! I tested it out with just one user on a team, and it works as expected. Should we change the heading to How does my _electricity_ bill compare to my neighbors in Cambridge? that way they don't add their June gas and electric bills and get a skewed score?

Maybe a score of 25.3 would be more user friendly than .25345653727353...

Team.rb line 27: ((last_month_total - this_month_total) / last_month_total)_.round(3) \* 100_

I ran RSpec and got a failing test in the user_spec. When I ran it again it passed. It must have been an edge case with the randomly generated bill amounts being more or less equal. The test should probably account for this as it is likely going to be the case somewhat frequently in production.

screen shot 2016-06-12 at 7 36 11 pm

**Comment by mzagaja** [comment]
I think you are right, a rounding function makes sense for the view. As far as the test failing, I think that based on your analysis the test itself is wrong, and that zero could be a valid result. (i.e. same energy bill over two months would be 0/something = 0). So that test should simply test for presence. I will make these changes tomorrow and update the PR.

File: app/models/team.rb [hunk 1]
```diff
@@ -12,8 +12,21 @@ class Team < ActiveRecord::Base
 
   # == Instance methods
   def score
-    total_score = users.map{ |u| u.score }.sum
+    bills = self.users.map { |user| user.most_recent_bills }
+    bills.reject { |score| !score.kind_of?(Array) | (score.length < 2) }
```
Comment (mzagaja) [inline app/models/team.rb]: I can probably tighten this up by having this logic go into most_recent_bills in user.rb and having it return a nil if it lacks two bills.
